util/repeatable_thread.h

```diff
@@ -34,7 +34,14 @@ class RepeatableThread {
         thread_([this] { thread(); }) {
   }
 
-  void cancel() {
+  // allow caller to pass a mutex and temporarily unlock the mutex during
+  // the process of cancel. Useful when function_ also need to lock the same
+  // mutex
+  void cancel(InstrumentedMutex* mutex = nullptr) {
```

Comment:
The `cancel` function looks a little strange with an extra `mutex` parameter. Is it necessary?